### PR TITLE
feat: add Time To Initial Display and Time To Full Display spans

### PR DESCRIPTION
**Pull request by stefanosiano**
This PR adds instructions on what Time To Initial Display and Time To Full Display spans are and how to set them up

### CODE REVIEW

File: src/docs/sdk/performance/time-to-initial-full-display.mdx [hunk 1]
```diff
@@ -0,0 +1,80 @@
+---
+title: "Time to Initial Display/Time to Full Display"
```
Comment (philipphofmann) [inline src/docs/sdk/performance/time-to-initial-full-display.mdx:2]: `m`: I like the title form the sidebar more.
```suggestion
title: "Time to Initial/Full Display"
```

File: src/docs/sdk/performance/time-to-initial-full-display.mdx [hunk 2]
```diff
@@ -0,0 +1,80 @@
...
+---
+
+We recommend implementing this feature for mobile and desktop SDKs.
+
+The current screen load transactions timings are not reliable to understand the time spent in the screen creation.
+This is because they wait for any span created in the meantime, including any file I/O or network request.
```
Comment (philipphofmann) [inline src/docs/sdk/performance/time-to-initial-full-display.mdx:8]: `l`:  A bit cleaner
```suggestion
The current screen load transactions timings are unreliable in understanding the time spent in the screen creation.
This is because they wait for any span created, including any file I/O or network request.
```

File: src/docs/sdk/performance/time-to-initial-full-display.mdx [hunk 3]
```diff
@@ -0,0 +1,80 @@
...
+Time to Initial Display and Time to Full Display are spans added to the screen load transactions,
+which can be used to show more accurate information on the time spent in the screen creation.
+Being spans, they don't impact on user quota.
+
+They start as soon as a new screen is created, as long as the screen load auto-generated transactions are enabled.
```
Comment (philipphofmann) [inline src/docs/sdk/performance/time-to-initial-full-display.mdx:13]: `m`: That is a bit unclear to me. I don't fully understand this sentence.

File: src/docs/sdk/performance/time-to-initial-full-display.mdx [hunk 4]
```diff
@@ -0,0 +1,80 @@
...
+
+## Time to Initial Display
+
+Time to Initial Display is a span with operation `ui.load.initial_display` which finishes automatically
+when the screen draws its first frame. It's controlled automatically by the SDK and doesn't
+require any manual intervention from the user.
+
+It's also set as a measurement of the transaction, with `time_to_initial_display` as key and the span duration as value.
+Note: The measurement is should be set only if the span finishes with an `OK`  status.
```
Comment (philipphofmann) [inline src/docs/sdk/performance/time-to-initial-full-display.mdx:22]: `m`: How could the TTID span end with a different status?

File: src/docs/sdk/performance/time-to-initial-full-display.mdx [hunk 5]
```diff
@@ -0,0 +1,80 @@
...
+
+## Time to Full Display
+
+Time to Full Display is a span with operation `ui.load.full_display` which is finished manually through
+the `Sentry.reportFullyDisplayed` static API. This feature has to be enabled by setting the SDK config
+option `enableTimeToFullDisplayTracing`. Remember that the screen load transactions wait for spans to finish,
+so the transaction will run until the user calls the API. So we recommend this feature to be disabled by default.
+It's also set as a measurement of the transaction, with `time_to_full_display` as key and the span duration as value.
+
+The following section explain the possible scenarios of the Time to Full Display.
```
Comment (philipphofmann) [inline src/docs/sdk/performance/time-to-initial-full-display.mdx:33]: `l`:
```suggestion
The following section explains the possible scenarios of the Time to Full Display.
```

File: src/docs/sdk/performance/time-to-initial-full-display.mdx [hunk 6]
```diff
@@ -0,0 +1,80 @@
...
+
+The specification is written in the [Gherkin syntax](https://cucumber.io/docs/gherkin/reference/).
+
+```Gherkin
+Scenario: Option disabled
+    Given an ongoing screen load/navigation transaction
+    Then the Full Display span is not started
```
Comment (philipphofmann) [inline src/docs/sdk/performance/time-to-initial-full-display.mdx:40]: `m`: Again please use the active voice. I'm going to make only one comment for the passive voice for the scenarios.
```suggestion
    Then the SDK doesn't start the Full Display span
```

File: src/docs/sdk/performance/time-to-initial-full-display.mdx [hunk 7]
```diff
@@ -0,0 +1,80 @@
...
+    Given an ongoing Full Display span
+    When the user calls the API `reportFullyDisplayed`
+    Then the SDK finishes the span
+    And sets the status to ok
+
+Scenario: API not called
+    Given an ongoing Full Display span
+    When the user doesn't call the API `reportFullyDisplayed` in 30 seconds
+    Then the SDK finishes the span
+    And sets the status to deadline_exceeded
```
Comment (philipphofmann) [inline src/docs/sdk/performance/time-to-initial-full-display.mdx:59]: `m`:
```suggestion
    And sets the status to deadline_exceeded
    And sets the timestamp to the timestamp of the Initial Display span
```

File: src/docs/sdk/performance/time-to-initial-full-display.mdx [hunk 8]
```diff
@@ -0,0 +1,80 @@
...
+
+Scenario: API called too early
+    Given an ongoing Full Display span and an ongoing Initial Display span
+    When the user calls the API `reportFullyDisplayed`
+    Then the SDK finishes the Full Display span
+    And
+    When the Initial Display span is finished
+    Then the SDK adjusts the Full Display span end timestamp to the Initial Display span end timestamp
```
Comment (philipphofmann) [inline src/docs/sdk/performance/time-to-initial-full-display.mdx:67]: `m`: Please also specify which status the SDK sets to both spans

File: src/docs/sdk/performance/time-to-initial-full-display.mdx [hunk 9]
```diff
@@ -0,0 +1,80 @@
...
+Scenario: API not called
+    Given an ongoing Full Display span
+    When the user doesn't call the API `reportFullyDisplayed` in 30 seconds
+    Then the SDK finishes the span
+    And sets the status to deadline_exceeded
+
+Scenario: API called too early
+    Given an ongoing Full Display span and an ongoing Initial Display span
+    When the user calls the API `reportFullyDisplayed`
+    Then the SDK finishes the Full Display span
```
Comment (philipphofmann) [inline src/docs/sdk/performance/time-to-initial-full-display.mdx:64]: `m`: What status does the SDK set?

File: src/docs/sdk/performance/time-to-initial-full-display.mdx [hunk 10]
```diff
@@ -0,0 +1,80 @@
...
+    And
+    When the Initial Display span is finished
+    Then the SDK adjusts the Full Display span end timestamp to the Initial Display span end timestamp
+
+Scenario: Another screen load/navigation transaction occurs
+    Given an ongoing Full Display span
+    When a new screen load/navigation transaction is started
+    Then the SDK finishes the Full Display span
+    And sets the status to deadline_exceeded
```
Comment (philipphofmann) [inline src/docs/sdk/performance/time-to-initial-full-display.mdx:73]: `m`: What is the value of the timestamp?

File: src/docs/sdk/performance/time-to-initial-full-display.mdx [hunk 11]
```diff
@@ -0,0 +1,80 @@
...
+    When the user calls the manual API `reportFullyDisplayed`
+    Then the SDK ignores the call
+
+Scenario: Option enabled
+    Given the enableTimeToFullDisplayTracing option is enabled
+    When the SDK creates a new screen load transaction
+    Then the Full Display span is started
+
+Scenario: Activity displayed
+    Given an ongoing Full Display span
```
Comment (philipphofmann) [inline src/docs/sdk/performance/time-to-initial-full-display.mdx:50]: `l`: A bit more specific
```suggestion
    Given an ongoing Full Display span
    And the SDK finished the  Initial Display span
```

File: src/docs/sdk/performance/time-to-initial-full-display.mdx [hunk 12]
```diff
@@ -0,0 +1,80 @@
...
+
+It's also set as a measurement of the transaction, with `time_to_initial_display` as key and the span duration as value.
+Note: The measurement is should be set only if the span finishes with an `OK`  status.
+
+
+## Time to Full Display
+
+Time to Full Display is a span with operation `ui.load.full_display` which is finished manually through
+the `Sentry.reportFullyDisplayed` static API. This feature has to be enabled by setting the SDK config
+option `enableTimeToFullDisplayTracing`. Remember that the screen load transactions wait for spans to finish,
```
Comment (philipphofmann) [inline src/docs/sdk/performance/time-to-initial-full-display.mdx:29]: `m`: Isn't it called `enableTimeToFullDisplay`?

**Review by philipphofmann — COMMENTED** [review]
Thanks for using the Gherkin syntax 👏 . It's a pleasure to read it and see all the edge cases.

File: src/docs/sdk/performance/time-to-initial-full-display.mdx (hunk 12 above)
Comment (stefanosiano) [reply src/docs/sdk/performance/time-to-initial-full-display.mdx:29]: on Android it's `io.sentry.traces.time-to-full-display.enable` in the manifest and `enableTimeToFullDisplayTracing` in the code, to be consistent with the other options currently available.
Guess we can rephrase it to `enableTimeToFullDisplay` in the docs anyway?

File: src/docs/sdk/performance/time-to-initial-full-display.mdx (hunk 3 above)
Comment (stefanosiano) [reply src/docs/sdk/performance/time-to-initial-full-display.mdx]: I rephrased it. Is it better now?

File: src/docs/sdk/performance/time-to-initial-full-display.mdx (hunk 4 above)
Comment (stefanosiano) [reply src/docs/sdk/performance/time-to-initial-full-display.mdx]: The case is when a new screen starts before showing the first frame
Like an Activity that is used to only start other activities

File: src/docs/sdk/performance/time-to-initial-full-display.mdx [hunk 13]
```diff
@@ -0,0 +1,85 @@
...
+title: "Time to Initial/Full Display"
+---
+
+We recommend implementing this feature for mobile and desktop SDKs.
+
+The current screen load transactions timings are unreliable in understanding the time spent in the screen creation.
+This is because they wait for any span created, including any file I/O or network request.
+Time to Initial Display and Time to Full Display are spans added to the screen load transactions,
+which can be used to show more accurate information on the time spent in the screen creation.
+Being spans, they don't impact on user quota.
```
Comment (philipphofmann) [inline src/docs/sdk/performance/time-to-initial-full-display.mdx:11]: `l`:
```suggestion
Being spans, they don't impact user quota.
```

File: src/docs/sdk/performance/time-to-initial-full-display.mdx (hunk 4 above)
Comment (philipphofmann) [reply src/docs/sdk/performance/time-to-initial-full-display.mdx]: Please add this essential piece of information to the docs.

File: src/docs/sdk/performance/time-to-initial-full-display.mdx [hunk 14]
```diff
@@ -0,0 +1,85 @@
...
+    And sets the status to deadline_exceeded
+    And sets the timestamp to the timestamp of the Initial Display span
+
+Scenario: API called too early
+    Given an ongoing Full Display span and an ongoing Initial Display span
+    When the user calls the API `reportFullyDisplayed`
+    Then the SDK finishes the Full Display span
+    And set the status to ok
+    And
+    When the Initial Display span is finished
```
Comment (philipphofmann) [inline src/docs/sdk/performance/time-to-initial-full-display.mdx:69]: `m`: You should only have one `When` per scenario. You split this scenario up into two.

File: src/docs/sdk/performance/time-to-initial-full-display.mdx [hunk 15]
```diff
@@ -0,0 +1,85 @@
...
+    When the user doesn't call the API `reportFullyDisplayed` in 30 seconds
+    Then the SDK finishes the span
+    And sets the status to deadline_exceeded
+    And sets the timestamp to the timestamp of the Initial Display span
+
+Scenario: API called too early
+    Given an ongoing Full Display span and an ongoing Initial Display span
+    When the user calls the API `reportFullyDisplayed`
+    Then the SDK finishes the Full Display span
+    And set the status to ok
```
Comment (philipphofmann) [inline src/docs/sdk/performance/time-to-initial-full-display.mdx:67]: `l`:
```suggestion
    And sets the status to ok
```

File: src/docs/sdk/performance/time-to-initial-full-display.mdx (hunk 12 above)
Comment (philipphofmann) [reply src/docs/sdk/performance/time-to-initial-full-display.mdx]: All the other options for controlling other automatic instrumentation have the `tracing` suffix. I think we should name the option `enableTimeToFullDisplayTracing`.

**Review by philipphofmann — COMMENTED** [review]
Almost LGTM

File: src/docs/sdk/performance/time-to-initial-full-display.mdx [hunk 16]
```diff
@@ -0,0 +1,88 @@
...
+
+Scenario: API called
+    Given an ongoing Full Display span
+    When the user calls the API `reportFullyDisplayed`
+    Then the SDK finishes the Full Display span
+    And set the status to ok
+
+Scenario: API called too early
+    Given an ongoing Initial Display span and the user calls the API `reportFullyDisplayed`
+    When the Initial Display span is finished
```
Comment (philipphofmann) [inline src/docs/sdk/performance/time-to-initial-full-display.mdx:72]:
```suggestion
    Given an ongoing Initial Display span
    And the user called the API `reportFullyDisplayed`
    When the SDK finishes the Initial Display span
```

**Review by philipphofmann — APPROVED** [review]
Minor comments, LGTM thanks @stefanosiano 👏
